news/_posts/2019-10-10-about-overriding-system.md

```diff
@@ -0,0 +1,131 @@
+
+> We can say that undoubtely, Overriding system was a bad idea and we
+> are deprecating it.
+
+### Overriding system deprecation in 1.7.x releases
+
+As most of you have noticed, it's not possible to override core classes located in the `src` folder: this is intended and not something we will implement/introduce back later.
+
+The override system is still available on "soon-to-be-migrated" pages, but we strongly discourage you to rely on it as when pages will be migrated, this feature won't be backported.
+
+The override system is deprecated in PrestaShop Front Office (and on the Customer Back Office): this will works until the end of 1.7.x and we will probably add some warnings when used to help developers to prepare the migration to the next major version.
+
+The documentation will be updated and we will remove every mention of overrides.
+
+### Overriding system in 1.8.x releases
+
+The roadmap of 1.8.0 is far from ready, but the very first major feature is the complete migration of the Front Office using Symfony Framework and Twig to get the same developer experience in both Front Office and Back Office.
+Overrides won't work and the override folder will be ignored.
+It's really too soon to think about it, but this probably means that the native PrestaShop autoloader won't be available in 1.8.x release, and we will depends on Composer only for the class autoloading.
+
+## How to override PrestaShop native behaviors?
+
+**The best and recommended way to alter PrestaShop behavior are the hooks for the data and the templating overriding system for the view.**
+
+Because we had the override system, we (the core developers) were lazy: after all, why should we care about extension points when developers can override almost everything in the classes?
+
+PrestaShop have a very long list of useful and not-that-useful hooks: since PrestaShop 1.7.3 we have introduced a new Hook Profiler to help developers figure out which hook(s) can help them to improve their shop and fits their customers needs.
+
+In PrestaShop 1.7.5 we have improved this profiler, making hooks executed and arguments available in the modules discoverables in a web interface and we have a lot of new ideas to make your experience with hooks enjoyable again.
+
+Also, Hooks are slow and we were less and less easy with the idea to add new hooks (ie add new extension points) in the core: isn't that weird in an extensible platform?
+
+In PrestaShop 1.7.5, we have started to work on a new architecture to manage hooks, on top of the Symfony Event Dispatcher component. This won't be available in 1.7 because we don't want to break hooks but the new system will allows us (and you) to add an infinity of hooks with a very low footprint on your shop performance.
```

Comment:
`This will be available in the next major release. We don't want to break hooks but the new system will allows everyone to add events as many as they wanted with a very low footprint on their shop performance.`